core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/ReadOnlyLDAPUserStoreManager.java

```diff
@@ -2865,8 +2865,14 @@ private void getExpressionConditionsAsList(Condition condition, List<ExpressionC
                 do {
                     List<String> tempUserList = new ArrayList<>();
                     answer = ldapContext.search(escapeDNForSearch(searchBase), searchFilter, searchControls);
+                    /*
+                    The size of the expression condition is used to verify the type of filter operation since the up
+                    coming steps needs to verify whether this is a multi attribute scenario or single attribute
+                    scenario.
+                     */
```

Comment:
This comment should move to the method that decides whether it's a single attribute case based on the list size

Reply:
Better if the condition list as a whole can be passed to the respective method and if it could resolve if it's a single attribute case

Reply:
fixed: 158a25c1b605c0b562a45162ae11eb54244a922e